src/main/java/lrbg/codriver/parser/Parser.java

```diff
@@ -0,0 +1,195 @@
+            }
+            int index = Integer.parseInt(arguments[1]);
+            return new UnmarkCommand(index);
+        }
+        case "todo":
+            if (arguments.length < 2) {
+                throw new CoDriverException("Error! You cannot provide a todo with no description!");
+            }
+            return new TodoCommand(commandLine.substring(5));
+        case "deadline": {
+            if (arguments.length < 2) {
+                throw new CoDriverException("Error! You cannot provide a deadline with no parameters!");
+            }
+
+            StringBuilder descriptionBuilder = new StringBuilder();
+            int i;
+            for (i = 1; i < arguments.length; i++) {
+                if (arguments[i].equals("/by")) {
+                    break;
+                }
+                descriptionBuilder.append(arguments[i]).append(" ");
+            }
+            if (descriptionBuilder.length() == 0) {
+                throw new CoDriverException("Error! You cannot provide a deadline with no description!");
+            }
+            descriptionBuilder.deleteCharAt(descriptionBuilder.length() - 1); // remove the last space
+
+            if (i >= arguments.length - 1) { // if the last word is /by or there is no /by
+                throw new CoDriverException("Error! You must provide a /by date for a deadline!");
+            }
+
+//        StringBuilder dateBuilder = new StringBuilder();
+            i++;
+            LocalDate date = null;
+            for (; i < arguments.length; i++) {
+                // check if in yyyy-mm-dd format
+                date = Parser.parseDate(arguments[i]);
+                if (date == null) {
+                    throw new CoDriverException("Error! The date provided must be in yyyy-mm-dd format!");
+                }
+//            dateBuilder.append(arguments[i]).append(" ");
+            }
+            if (date == null) {
+                throw new CoDriverException("Error! You must provide a /by date for a deadline!");
+            }
+            return new DeadlineCommand(descriptionBuilder.toString(), date);
+        }
+        case "event": {if (arguments.length < 2) {
+            throw new CoDriverException("Error! You cannot provide an event with no parameters!");
+        }
+
+            StringBuilder descriptionBuilder = new StringBuilder();
+            int i;
+            for (i = 1; i < arguments.length; i++) {
+                if (arguments[i].equals("/from")) {
+                    break;
+                }
+                descriptionBuilder.append(arguments[i]).append(" ");
+            }
+            if (descriptionBuilder.length() == 0) {
+                throw new CoDriverException("Error! You cannot provide an event with no description!");
+            }
+            descriptionBuilder.deleteCharAt(descriptionBuilder.length() - 1); // remove the last space
+
```

Comment:
The Java Coding Standard that we are referring suggests that:

> Logical units within a block should be separated by one blank line.

Hence, perhaps you may want to insert some blank lines to improve the code readability?